lib/DxilDia/DxcPixDxilDebugInfo.cpp

```diff
@@ -172,32 +172,24 @@ dxil_debug_info::DxcPixDxilInstructionOffsets::DxcPixDxilInstructionOffsets(
 {
   assert(SourceColumn == 0);
   (void)SourceColumn;
-
-  auto files = pSession->Contents()->operands();
-  for (const auto& file : files)
-  {
-    auto candidateFilename = llvm::dyn_cast<llvm::MDString>(file->getOperand(0))
-        ->getString();
-
-    if (CompareFilenames(FileName, candidateFilename.str().c_str()))
-    {
-
-      auto Fn = pSession->DxilModuleRef().GetEntryFunction();
-      auto &Blocks = Fn->getBasicBlockList();
-      for (auto& CurrentBlock : Blocks) {
-        auto& Is = CurrentBlock.getInstList();
-        for (auto& Inst : Is) {
-          auto & debugLoc = Inst.getDebugLoc();
-          if (debugLoc)
+  auto Fn = pSession->DxilModuleRef().GetEntryFunction();
+  auto &Blocks = Fn->getBasicBlockList();
+  for (auto& CurrentBlock : Blocks) {
+    auto& Is = CurrentBlock.getInstList();
+    for (auto& Inst : Is) {
+      auto & debugLoc = Inst.getDebugLoc();
+      if (debugLoc)
+      {
+        unsigned line = debugLoc.getLine();
+        if (line == SourceLine)
```

Comment:
Is SourceLine the actual line number in the preprocessed shader or the line in the #line directive?

Reply:
It's the line (and filename) from the #line directive

Reply:
I thought the lines in the DebugLoc are actually lines in the flattened preprocessed file, wouldn't that not match with the lines from #line?

Reply:
More accurately, if the trace is from an older compiler (like, e.g. the UE4 shaders I've been working with), then it's the #lines. If it's a new compiler, or a recompiled (PIX edit-and-continue) one, then it's the preprocessed file.

Reply:
So for new compiler output, this change doesn't make any difference because the older outer loop over filenames should be the same as the one from the debugloc

Reply:
Okay gotcha. Thank you!